website/docs/index.md

```diff
@@ -53,6 +53,41 @@ ln -s -f ../../scripts/pre-commit .git/hooks/pre-commit
 ```
 
 
+## How to Create a Good UNIX Interface
+
+One of our goals with this project is learning how to write programs and, more importantly,
+the importance of not writing code vs. writing code. 
+
+Here are some 
+- [good rules for writing good unix programs](https://lasr.cs.ucla.edu/ficus-members/geoff/interfaces.html)
+
+A good example can be seen in notions of progress bars. The idea of progress bars has come up twice now, from 
+two different contributors, once in cp, and once in dd.
```

Comment:
So now let's take a more interesting example.

tar | dd | dd | dd 

and we want to monitor progress of all of them. How do we do that? 

If we have expvars, we might be able to do something like this:
PROGRESS=progress commands ...
i.e. we could arrange for a command to kick off all the commands, and we could further state that all their progress should be monitored in one place. This is simply not possible to do with a command line option that dumps stuff to stderr: which one of the dd commands do you want to watch? And how will you tell them apart? 

If we can get a reasonable expvars-like approach we can start to do things that are otherwise very hard.